src/util/asmap.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <vector>
+#include <assert.h>
+
+int DecodeBits(std::vector<bool> stream, int* bitpos, int minval, const int bit_sizes[], int bit_sizes_n)
```

Comment:
A more C++ish way to do this would be to pass a `std::vector<bool>::const_iterator& it` (for the beginning of the value, updated to be after the value afterwards) and a `const std::vector<bool>::const_iterator& end` (for the end of the area).